libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -436,6 +446,32 @@ bool AC_Fence::check_fence_alt_min()
     return false;
 }
 
+// returns true if alt-frame is converted successfully
+bool AC_Fence::conv_alt_frame()
+{
+    if (_alt_frame != _alt_frame_ext) {
+        int32_t _new_max_alt_cm, _new_min_alt_cm;
+        if (fenceloc.get_spec_alt_cm(static_cast<Location::AltFrame>(int(_alt_frame)), Location::AltFrame((int)_alt_frame_ext), _alt_max * 100.0f, _new_max_alt_cm)) {
```

Comment:
You should be able to use this new method without having the location variable, maybe `Location::get_spec_alt_cm` * I think